practice_2/AnimalsApp/Services/NetworkService/NetworkService.swift

```diff
@@ -1,42 +1,85 @@
 import Foundation
 
 protocol NetworkServiceProtocol {
+    // Dog Api https://dog.ceo/dog-api/
     func getAllDogBreeds(completion: @escaping (Result<DogBreeds, Error>) -> Void)
     func getRandomDogImage(by breed: String, completion: @escaping (Result<DogRamdomImage, Error>) -> Void)
     func getAllDogImages(by breed: String, completion: @escaping (Result<DogImages, Error>) -> Void)
+
+    // Cat Api https://dog.ceo/dog-api/
+    func getAllCatBreeds(completion: @escaping (Result<CatBreeds, Error>) -> Void)
+    func getCatImages(by breedId: String, in quantity: Int, completion: @escaping (Result<CatImages, Error>) -> Void)
 }
 
 enum AnimalsApiType {
+    static let scheme = "https"
+    // Dog Api
     case getAllDogBreeds
     case getRandomDogImage
     case getAllDogImages
 
-    var baseUrl: String {
-        "https://dog.ceo/api/"
+    // Cat Api
+    case getAllCatBreeds
+    case getCatImages
+
+    var host: String {
+        switch self {
+        case .getAllDogBreeds, .getRandomDogImage, .getAllDogImages:
+            return "dog.ceo"
+        case .getAllCatBreeds, .getCatImages:
+            return "api.thecatapi.com"
+        }
     }
     var path: String {
         switch self {
         case .getAllDogBreeds:
-            return "breeds/list/all"
+            return "/api/breeds/list/all"
         case .getRandomDogImage:
-            return "breed/{BREED}/images/random"
+            return "/api/breed/{BREED}/images/random"
         case .getAllDogImages:
-            return "breed/{BREED}/images"
+            return "/api/breed/{BREED}/images"
+        case .getAllCatBreeds:
+            return "/v1/breeds"
+        case .getCatImages:
+            return "/v1/images/search"
         }
     }
-    func getRequest(urlParams: [String: String]?) -> URLRequest {
+
+    private func createUrl(urlParams: [String: String]?) -> URL {
         var path = path
         if let breed = urlParams?["breed"] {
             path = path.replacingOccurrences(of: "{BREED}", with: breed)
         }
 
-        let url = URL(string: path, relativeTo: URL(string: baseUrl)!)!
+        var urlComponents = URLComponents()
+        urlComponents.scheme = AnimalsApiType.scheme
+        urlComponents.host = host
+        urlComponents.path = path
+
+        if let breedId = urlParams?["breedId"], let limit = urlParams?["limit"] {
+            urlComponents.queryItems = [
+                URLQueryItem(name: "size", value: "small"),
+                URLQueryItem(name: "limit", value: "\(limit)"),
+                URLQueryItem(name: "breed_id", value: "\(breedId)")
+            ]
+        }
+
+        print("url: \(urlComponents.url!.absoluteString)")
+        return URL(string: urlComponents.url!.absoluteString)!
```

Comment:
avoid force unwrap optionals
please use swiftlint and correct all warnings and errors

Reply:
Done